### PR TITLE
Improved getter/setter property to recognize setters that return bools.

### PR DESCRIPTION
**Pull request by tritao**

### CODE REVIEW

File: tests/Common/Common.Tests.cs [hunk 1]
```diff
@@ -473,6 +473,8 @@ public void TestProperties()
 
         prop.GetterAndSetterWithTheSameName = 25;
         Assert.That(prop.GetterAndSetterWithTheSameName, Is.EqualTo(25));
+
+        prop.SetterReturnsBoolean(10);
```
Comment (ddobrev) [inline tests/Common/Common.Tests.cs]: What is this? Shouldn't it have been turned into a property?

File: src/Generator/Passes/GetterSetterToPropertyPass.cs [hunk 2]
```diff
@@ -247,7 +249,9 @@ private static string GetPropertyNameFromSetter(Method setter)
 
             private void DistributeMethod(Method method)
             {
-                if (method.OriginalReturnType.Type.IsPrimitiveType(PrimitiveType.Void))
+                if (method.OriginalReturnType.Type.IsPrimitiveType(PrimitiveType.Void) ||
```
Comment (ddobrev) [inline src/Generator/Passes/GetterSetterToPropertyPass.cs]: Extract method.OriginalReturnType.Type to a local and desugar it.

File: src/Generator/Passes/GetterSetterToPropertyPass.cs [hunk 3]
```diff
@@ -177,6 +177,8 @@ private static void GenerateProperty(string name, DeclarationContext context, Me
                 }
                 property.GetMethod = getter;
                 property.SetMethod = setter;
+                if (setter != null && !setter.ReturnType.Type.IsPrimitiveType(PrimitiveType.Void))
+                    setter.ReturnType = new QualifiedType(new BuiltinType(PrimitiveType.Void));
```
Comment (ddobrev) [inline src/Generator/Passes/GetterSetterToPropertyPass.cs]: Wouldn't this negatively affect the internal return type and therefore the P/Invocation? Does it work with an indirect return param?

File: src/Generator/Passes/GetterSetterToPropertyPass.cs [hunk 4]
```diff
@@ -247,7 +249,9 @@ private static string GetPropertyNameFromSetter(Method setter)
 
             private void DistributeMethod(Method method)
             {
-                if (method.OriginalReturnType.Type.IsPrimitiveType(PrimitiveType.Void))
+                if (method.OriginalReturnType.Type.IsPrimitiveType(PrimitiveType.Void) ||
+                    (method.OriginalReturnType.Type.IsPrimitiveType(PrimitiveType.Bool) &&
+                     method.Parameters.Count > 1))
```
Comment (ddobrev) [inline src/Generator/Passes/GetterSetterToPropertyPass.cs]: More than 1 parameter? How is this supposed to work with setters only having one?

**Comment by tritao** [comment]
Closing, fixed by https://github.com/mono/CppSharp/pull/917.

**Pull request closed without merging**